Objects/typeobject.c

```diff
@@ -3506,19 +3507,18 @@ mro_check(PyTypeObject *type, PyObject *mro)
     for (i = 0; i < n; i++) {
         PyObject *obj = PyTuple_GET_ITEM(mro, i);
         if (!PyType_Check(obj)) {
-            PyErr_Format(
-                PyExc_TypeError,
-                "mro() returned a non-class ('%.500s')",
-                Py_TYPE(obj)->tp_name);
+            PyErr_Format(PyExc_TypeError,
+                         "%s.mro() returned a non-class ('%T')",
+                         type->tp_name, obj);
             return -1;
         }
         PyTypeObject *base = (PyTypeObject*)obj;
 
         if (!is_subtype_with_mro(lookup_tp_mro(solid), solid, solid_base(base))) {
             PyErr_Format(
                 PyExc_TypeError,
-                "mro() returned base with unsuitable layout ('%.500s')",
-                base->tp_name);
+                "%s.mro() returned base with unsuitable layout ('%.500s')",
```

Comment:
I guess there were good reasons to use `%s` instead of `%N`, right?

Reply:
Oh, if only I could remember why I did this myself :)